ext/opcache/shared_alloc_posix.c

```diff
@@ -51,7 +78,12 @@ static int create_segments(size_t requested_size, zend_shared_segment_posix ***s
 	(*shared_segments_p)[0] = shared_segment;
 
 	sprintf(shared_segment_name, "/ZendAccelerator.%d", getpid());
-	shared_segment->shm_fd = shm_open(shared_segment_name, O_RDWR|O_CREAT|O_TRUNC, 0600);
+	shared_segment->shm_fd = 
+#if defined(HAVE_SHM_CREATE_LARGEPAGE)
+                shared_segment_lg_index > 0 ?
+                shm_create_largepage(shared_segment_name, shared_segment_flags, shared_segment_lg_index, SHM_LARGEPAGE_ALLOC_DEFAULT, shared_segment_mode) :
+#endif
+	        shm_open(shared_segment_name, shared_segment_flags, shared_segment_mode);
```

Comment:
May be it makes sense to fallback to ``shm_open`` if ``shm_create_largepage`` is failed?

Reply:
indeed